spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolverAutoConfiguration.java

```diff
@@ -80,4 +114,74 @@ public ApplicationConfigurationMetadataResolver metadataResolver(
 			DefaultContainerImageMetadataResolver containerImageMetadataResolver) {
 		return new BootApplicationConfigurationMetadataResolver(containerImageMetadataResolver);
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public RestTemplate metadataRestTemplate(RestTemplateBuilder builder, ContainerImageMetadataProperties properties)
+			throws NoSuchAlgorithmException, KeyManagementException {
+
+		StringHttpMessageConverter octetToStringMessageConverter = new StringHttpMessageConverter();
+		List<MediaType> mediaTypeList = new ArrayList(octetToStringMessageConverter.getSupportedMediaTypes());
+		mediaTypeList.add(MediaType.APPLICATION_OCTET_STREAM);
+		octetToStringMessageConverter.setSupportedMediaTypes(mediaTypeList);
+
+		if (!properties.isDisableSslVerification()) {
+			return builder.additionalMessageConverters(octetToStringMessageConverter).build();
+		}
+
+		// From https://www.javacodemonk.com/disable-ssl-certificate-check-resttemplate-e2c53583
+
+		// Trust manager that blindly trusts all SSL certificates.
+		TrustManager[] trustAllCerts = new TrustManager[] {
+				new X509TrustManager() {
+					public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+						return new X509Certificate[0];
+					}
+
+					public void checkClientTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+					}
+
+					public void checkServerTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+					}
+				}
+		};
+		SSLContext sslContext = SSLContext.getInstance("SSL");
+		// Install trust manager to SSL Context.
+		sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+		// Create an HttpClient that uses the custom SSLContext and do not verify cert hostname.
+		CloseableHttpClient httpClient = HttpClients.custom().setSSLContext(sslContext)
+				.setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE).build();
+		HttpComponentsClientHttpRequestFactory customRequestFactory = new HttpComponentsClientHttpRequestFactory();
+		customRequestFactory.setHttpClient(httpClient);
+		// Create a RestTemplate that uses custom request factory
+		return builder.requestFactory(() -> customRequestFactory)
+				.additionalMessageConverters(octetToStringMessageConverter).build();
+
+	}
+
+	@Value("${.dockerconfigjson}")
+	public String dockerconfigjson;
+
+	@Bean
+	public Map<String, RegistryConfiguration> registryConfigurationMap(ContainerImageMetadataProperties properties) {
+
+		logger.info("dockerconfigjson  >>>> :" + dockerconfigjson);
```

Comment:
this may contain base64 encoded things one may not want to have in logs

Reply:
Right it does and they are even base64 decoded :) will remove all log entires  that may revel registry configurations and secrets.